### ISSUE TITLE
"can't compare datetime.datetime to datetime.date" while "draft_if_future_date: true"

### ISSUE THREAD

**Issue by whallin** [squidfunk/mkdocs-material#5618]
### Context

I'm setting up the blog plugin to work how I want it to. For example, "scheduled" blog posts should always be marked as drafts by default. But this isn't quite working the way it should, based on how the official documentation describes how it should work, at least.

### Bug description

While `draft_if_future_date` is set to `true` in the blog plugin, the build always seems to fail with the error `TypeError: can't compare datetime.datetime to datetime.date`.

### Related links

- [Setting up a blog: Drafts](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#drafts)

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11781693/example.zip)

### Steps to reproduce

1. Setup the blog plugin following the [official documentation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/)
2. Set the [`draft_if_future_date`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#+blog.draft_if_future_date) property to `true`
3. Build the site using `mkdocs build` (with the info plugin disabled, of course)

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and for the excellent reproduction. Fixed in [`bd010db`](https://github.com/squidfunk/mkdocs-material-insiders/commit/bd010db904c636dc092d357f251b1489728c974c). Might be a regression, as this worked before. I made sure that it now works with `date`, as well as `datetime` values, e.g.:

``` yaml
date: 2023-08-01 # date
date: 2023-08-01T12:00:00 # datetime
```

**Comment by squidfunk** [comment]
Released as part of [9.1.17+insiders-4.36.1](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/9.1.17-insiders-4.36.1)!
